src/components/CardContainer/CardContainer.tsx

```diff
@@ -31,7 +30,10 @@ class CardContainer extends React.Component<MyProps, MyState> {
   };
 
   render() {
-    const gameCards = this.state.games.map((game) => {
+    const filteredBySearch = this.state.games.filter((game) => {
+      return game.name.toLowerCase().includes(this.props.query.toLowerCase());
+    });
+    const gameCards = filteredBySearch.map((game) => {
       return <GameCard key={game.id + game.slug} {...game} />;
     });
     return <section className="game-card-container">{gameCards}</section>;
```

Comment:
I see html handles only happy path and if there is at least 1 game card. So, it doesn't cover load or error states.